File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/NightMessages/Roleblocked.java
package townofsalemcalculator.Conditions.Concrete.NightMessages;

import java.util.Arrays;
import java.util.List;
import townofsalemcalculator.Conditions.RoleGroup.CompositeRoleGroup;
import static townofsalemcalculator.Role.Consort;
import static townofsalemcalculator.Role.Escort;

/**
 * The Concrete Condition that you were roleblocked
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-28
 */
public class Roleblocked extends NightMessage {
    
    public Roleblocked() {
        super(new RoleBlockerGroup());
    }
    
    @Override
    public String getCondition() {
        return "Someone occupied your night. You were role blocked";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"Rbed", "Roleblocked", "Role", "Blocked", "Occupied", "You", "Someone"});
        keyWords.addAll(Escort.getKeyWords());
        keyWords.addAll(Consort.getKeyWords());
        return keyWords;
    }
    
    static class RoleBlockerGroup extends CompositeRoleGroup {    
        public RoleBlockerGroup() {
            addRoleGroup(new townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup(Escort));
            addRoleGroup(new townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup(Consort));
        }
    }
}

File: src/Conditions/Concrete/Effects/Roleblocked.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class Roleblocked(ExistenceCondition):
    """ The Roleblocked condition is used when during the night you were roleblocked (either by an Escort or Consort),
    i.e. you received the message: 'Someone occupied your night. You were role blocked' """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.ESCORT, Role.CONSORT}), amnesiacRemembered)